crates/zk-por-core/src/recursive/prove.rs

```diff
@@ -0,0 +1,64 @@
+use plonky2::{
+    iop::witness::{PartialWitness, WitnessWrite},
+    plonk::{
+        circuit_data::{CircuitData, CommonCircuitData, VerifierOnlyCircuitData},
+        config::{AlgebraicHasher, GenericConfig},
+        proof::ProofWithPublicInputs,
+        prover::prove,
+    },
+    util::timing::TimingTree,
+};
+
+use crate::types::{D, F};
+use anyhow::Result;
+
+use super::circuit::RecursiveTargets;
+
+pub type ProofTuple<F, C, const D: usize> =
+    (ProofWithPublicInputs<F, C, D>, VerifierOnlyCircuitData<C, D>, CommonCircuitData<F, D>);
+
+pub fn prove_n_subproofs<
+    C: GenericConfig<D, F = F>,
+    InnerC: GenericConfig<D, F = F>,
+    const N: usize,
+>(
+    sub_proofs: Vec<ProofWithPublicInputs<F, InnerC, D>>,
```

Comment:
The reason I added this generic was to allow us to use arrays of length N rather than vecs, (allows shallow copy rather than clone). If we are not using arrays of fixed length you can remove the generic.

Reply:
we are using this generic in return value type.